image_writer.py

```diff
@@ -0,0 +1,80 @@
+"""Takes in image text (top text and bottom text) and an image, and returns an image with text written on it in meme
+style.
+"""
+
+from PIL import Image, ImageDraw, ImageFont
+from random import randint
+
+
+def write_text_to_image(text_top, text_bottom, image_name):
+    """Main function. Writes inputted text to image in 'meme' style"""
+    image = _import_image(image_name)
+    # processed_image = _process_image
+    font, text_color, spacing = _determine_text_attributes()
+    draw = ImageDraw.Draw(image)
+    text_top_x, text_top_y, text_bottom_x, text_bottom_y = _determine_text_placement(draw, image, font, text_top,
+                                                                                     text_bottom)
+    _draw_image(draw, text_top, font, text_color, text_top_x, text_top_y, spacing)
+    _draw_image(draw, text_bottom, font, text_color, text_bottom_x, text_bottom_y, spacing)
+    _save_image(image, image_name)
+
+
+def _import_image(image_name):
+    """Imports selected image."""
+    return Image.open(image_name)
+
+
+# def _process_image(image):
+#     """Eventually will hold randomized image processing.
+#     """
+#     return image
+
+
+def _determine_text_attributes():
+    """Sets attributes for proper 'meme' style text"""
+    font = ImageFont.truetype('impact.ttf', 60)
+    text_color = (255, 255, 255)
+    spacing = 10
+    return font, text_color, spacing
+
+
+def _determine_text_placement(draw, image, font, text_top, text_bottom):
+    """Measures image and text blocks and determines where to place each."""
+    image_w, image_h = _measure_image(image)
+    text_top_w, text_top_h = _measure_text(draw, text_top, font)
+    text_bottom_w, text_bottom_h = _measure_text(draw, text_bottom, font)
+    text_top_x, text_top_y = image_w // 2 - text_top_w // 2, image_h // 16 - text_top_h // 2
+    text_bottom_x, text_bottom_y = image_w // 2 - text_bottom_w // 2,  image_h // 12 * 11 - text_bottom_h // 2
+    return text_top_x, text_top_y, text_bottom_x, text_bottom_y
```

Comment:
If you split this function into a few smaller steps, you can actually doctest it.

1. Get image dims.
1. Get text dims.
1. From dims, calculate coords.

That last step is the meat of the matter and can be doctested since it's just numbers in and out!